crates/optimism/primitives/src/transaction/signed.rs

```diff
@@ -213,11 +214,11 @@ impl OpTransaction for OpTransactionSigned {
     }
 }
 
-impl FillTxEnv<revm_optimism::OpTransaction<TxEnv>> for OpTransactionSigned {
-    fn fill_tx_env(&self, tx_env: &mut revm_optimism::OpTransaction<TxEnv>, sender: Address) {
-        let envelope = self.encoded_2718();
+impl FromRecoveredTx<OpTransactionSigned> for revm_optimism::OpTransaction<TxEnv> {
```

Comment:
here we ideally also need support for `WithEncoded`, but we're running into dep graph issues here I think